### ISSUE TITLE
Cannot compile with a cgo package

### ISSUE THREAD

**Issue by sungwoncho** [goreleaser/goreleaser#795]
**Describe the bug**

Compiling with a cgo package succeeds with `go build` but fails when done through goreleaser.

**To Reproduce**

[This repo](https://github.com/sungwoncho/goreleaser-issue) contains a minimal reproduction.

**Expected behavior**

Goreleaser should compile if passed the same arguments and flags that are passed to `go build`

**Environment (please complete the following information):**

- OS: mac
- OS version: Darwin MIKEs-MacBook-Air.local 16.6.0 Darwin Kernel Version 16.6.0: Fri Apr 14 16:21:16 PDT 2017; root:xnu-3789.60.24~6/RELEASE_X86_64 x86_64
- GoReleaser Version 0.84.0, commit 93a0055d036200749a25b5a3454737b0de6af2d3, built at 2018-08-28T14:54:55Z

**Additional context**
Please refer to the reproduction repository.

**Comment by sungwoncho** [comment]
Thanks for the clarification.

**Comment by caarlos0** [comment]
No problem, sorry it doesnt work in your case :/

**Comment by sgnn7** [comment]
@caarlos0 What's your guys' take on adding CGO support in the public images? I think I can get a PR for that pretty easily.

**Comment by caarlos0** [comment]
@sgnn7 it's not that simple I believe, take a look at comments on https://github.com/goreleaser/goreleaser/issues/708

**Comment by caarlos0** [comment]
> (for Linux builds only though).

yeah, that's the deal... I guess we can have 2 docker images then, one with CGO support for linux only and other "clean" (the clean being the default)... what do you think?

**Comment by sgnn7** [comment]
@caarlos0 👍 yeah whatever you think is best. I don't think it bulks up the image too bad if it's in the default one I think (and would reduce confusion if there's only one image) but I can see arguments on both sides of this 🤷‍♂️ . Just let me know how you would like the changes to go and I'll see if I can help out.

**Comment by caarlos0** [comment]
I think 2 images is best, if you can open the PR it would be awesome! Thanks

**Comment by sgnn7** [comment]
@caarlos0 Sounds good - do you want me just to create something like `Dockerfile.cgo`?

**Comment by caarlos0** [comment]
sounds great :)

**Comment by jimafisk** [comment]
@sgnn7 thanks for adding cgo build support! Can you advise me on how I would go about using this? I updated my `.goreleaser.yml` to enable CGO:
```yml
builds:
- env:
  - CGO_ENABLED=1
```
But my build fails with the following error:
```
   ⨯ release failed after 43.50s error=failed to build for linux_386: # runtime/cgo
In file included from _cgo_export.c:3:0:
##[error]/usr/include/stdlib.h:25:10: fatal error: bits/libc-header-start.h: No such file or directory
 #include <bits/libc-header-start.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

##[error]The process '/opt/hostedtoolcache/goreleaser-action/0.140.1/x64/goreleaser' failed with exit code 1
```

You can see the full broken build here: https://github.com/plentico/plenti/runs/946698117#step:8:130

Is it obvious what I'm doing wrong? Thanks you!

**Comment by sgnn7** [comment]
Hey @jimafisk,
Sadly, the GH action that you're using is just downloading goreleaser binaries and running things. I don't know much about it but I bet it's the vanilla goreleaser and the support for cross compilation is specifically in the docker images with `-cgo` suffix. If this is the case, you may need to manually use the goreleaser docker image instead of `goreleaser/goreleaser-action`.
